plugin/check_extra_field_author_company/CheckExtraFieldAuthorsCompanyPlugin.php

```diff
@@ -206,23 +270,60 @@ public function SaveAuthorsField()
         $data['default_value'] = null;
         $data['variable'] = 'authors';
         $data['visible'] = 1;
-        $data['display_text'] = strtolower(Database::escape_string(Security::remove_XSS($data['display_text'])));
+        $data['display_text'] = strtolower($data['display_text']);
         $schedule = new ExtraField('lp');
         $schedule->save($data);
     }
 
+    /**
+     *  Set Yes or Not selector for authorlp field.
+     *
+     * @param $authorLpId
+     */
+    public function setYesNoToAuthor($authorLpId)
+    {
+        $options = [
+            0 => 'No',
+            1 => 'Yes',
+        ];
+        $order = 0;
+        $authorId = (int) $authorLpId;
+        if ($authorId != 0) {
+            $extraFieldValueUser = new ExtraFieldOption('user');
+            $items = $extraFieldValueUser->get_field_options_by_field($authorLpId);
+            foreach ($items as $item) {
+                if (isset($options[0]) && (isset($item['option_value']) == $options[0] || isset($item['option_value']) == $options[1])) {
+                    unset($options[$item['option_value']]);
+                    $order++;
+                }
+            }
+
+            for ($i = 0; $i < count($options); $i++) {
+                $extraFieldOptionValue = $options[$i];
+                $fieldOption = new ExtraFieldOption('user');
+                $fieldOption->saveOptions([
+                    'field_id' => $authorLpId,
+                    'option_value' => $order,
+                    'display_text' => $extraFieldOptionValue,
+                    'option_order' => $order,
+                ]);
+                $order++;
+            }
+        }
+    }
+
     /**
      * Remove the extra fields set by the plugin.
      */
     public function uninstall()
     {
         $companyExist = $this->CompanyFieldExist();
         if ($companyExist == true) {
-            $this->removeCompanyField();
+            // $this->removeCompanyField();
```

Comment:
Why is this line commente?

Reply:
It is prevented to delete the changes created so as not to delete the records. Although it could be prevented by hiding the fields. Would it be better to hide the fields instead of deleting the records? @ywarnier